### DISCUSSION TITLE
No communication anymore

### DISCUSSION THREAD

**Discussion by johanvh83** [wlcrs/huawei_solar#330] [Setting up the integration]
Integration works for months without problems. Was 1 week away for holidays, came back and last 3 days no info about solar system. Checked on FusionSolar (Huawei portal) and that shows me data so the solar system works fine.
HACS had new integration release and HA Core was also outdated. Upgraded both but no solution.
Checked logs:

Logger: huawei_solar.huawei_solar
Source: /usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py:296
First occurred: 1:08:21 PM (26 occurrences)
Last logged: 1:11:46 PM

Modbus client is not connected to the inverter.
NoneType: None

Checked Inverters local IP but it remained the same. Restarting HA and re-loading the integration doesn't solve the problem.

I removed the integration (used latest beta version). Restarted HA and installed the newest stable version 1.5.2
When I have to entered the IP adress and the integration starts communication it gives 'Failed to connect'.
I restarted my Wifi network and the inverter. Waited 5 minutes to initialise and tried once more but same problem.
I'm double checked the entered IP address, this is correct.
Checked firmware from inverter; not changed: V200R001C00SPC130
Could anyone tell me how to start debugging? Anything to check on FusionSolar portal?
I don't know how to proceed.
Any idea welcome to resolve this issue

**Comment by galtamur** [comment]
I have the same problem. I tried with nmap and it shows the tcp port of the inverter is closed. I contacted the Huawei support and they told be there has been an upgrade on the inverter which disabled the communication via MODBUS; he told me to buy the TCP Dongle to re-enable it. Hope that's true!

**Reply by johanvh83** [reply]
Frustrating! This worked for a long time and now, without any notice, the manufacturer changes the way the equipment works.
Is this allowed?
Do you believe the installer can re-activate the communication via MODBUS as before?
How do you contact Huawei support? Ready to send a complain!

**Comment by galtamur** [comment]
I just phoned to the support.

You can find numbers here: https://solar.huawei.com/it/service-support/tsc

 

They told me the previously working communication can only be reestablished by installing the dongle…

**Comment by johanvh83** [comment]
The inverter has also a communication port. Accoring https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-the-inverter it should be able to read the same data on terminals 1 and 2 of this port. Anyone tried already?
I don't believe I'm the only 'victom' of this Huawei change.

**Comment by galtamur** [comment]
Yes there are two ways for communicating, serial or network: two different dongles are required for each.

I think the easiest way (and closest to the old communication method working till last week) is to use the network dongle.

**Comment by johanvh83** [comment]
If Huawei thinks to earn money by disable a feature that requires now a dongle to enable it again, they are wrong.
I bought this serial RS485 to Wifi converter: Elfin EW-11 [https://nl.aliexpress.com/item/32916950283.html?spm=a2g0o.order_list.order_list_main.10.2fd779d2Urgxqv&gatewayAdapt=glo2nld](url)
